sdk/core/core-rest-pipeline/src/nodeHttpClient.ts

```diff
@@ -229,7 +230,7 @@ class NodeHttpClient implements HttpClient {
           req.end(ArrayBuffer.isView(body) ? Buffer.from(body.buffer) : Buffer.from(body));
         } else {
           logger.error("Unrecognized body type", body);
-          throw new RestError("Unrecognized body type");
+          reject(new RestError("Unrecognized body type"));
```

Comment:
Ah, we used to run into this in event hubs/service bus a lot too.

Reply:
This one shouldn't actually hurt us because throwing from inside the promise constructor (when sync) is handled correctly, but it didn't look as nice.